Formula/mumps.rb

```diff
@@ -0,0 +1,201 @@
+                    "FL=#{ENV["FC"]} -fPIC"]
+    end
+
+    if build.with? "openblas"
+      make_args << "LIBBLAS=-L#{Formula["openblas"].opt_lib} -lopenblas"
+    elsif build.with? "veclibfort"
+      make_args << "LIBBLAS=-L#{Formula["veclibfort"].opt_lib} -lvecLibFort"
+    else
+      make_args << "LIBBLAS=-lblas -llapack"
+    end
+
+    ENV.deparallelize # Build fails in parallel on Mavericks.
+
+    system "make", "alllib", *(shlibs_args + make_args)
+
+    lib.install Dir["lib/*"]
+    lib.install "libseq/libmpiseq.dylib" if build.without? "mpi"
+
+    # Build static libraries (e.g., for Dolfin)
+    system "make", "alllib", *make_args
+    (libexec/"lib").install Dir["lib/*.a"]
+    (libexec/"lib").install "libseq/libmpiseq.a" if build.without? "mpi"
+
+    inreplace "examples/Makefile" do |s|
+      s.change_make_var! "libdir", lib
+    end
+
+    libexec.install "include"
+    include.install_symlink Dir[libexec/"include/*"]
+    # The following .h files may conflict with others related to MPI
+    # in /usr/local/include. Do not symlink them.
+    (libexec/"include").install Dir["libseq/*.h"] if build.without? "mpi"
+
+    doc.install Dir["doc/*.pdf"]
+    pkgshare.install "examples"
+
+    prefix.install "Makefile.inc"  # For the record.
+    File.open(prefix/"make_args.txt", "w") do |f|
+      f.puts(make_args.join(" "))  # Record options passed to make.
+    end
+
+    if build.with? "mpi"
+      resource("mumps_simple").stage do
+        simple_args = ["CC=#{ENV["MPICC"]}", "prefix=#{prefix}", "mumps_prefix=#{prefix}",
+                       "scalapack_libdir=#{Formula["scalapack"].opt_lib}"]
+        if build.with? "scotch5"
+          simple_args += ["scotch_libdir=#{Formula["scotch5"].opt_lib}",
+                          "scotch_libs=-L$(scotch_libdir) -lptesmumps -lptscotch -lptscotcherr"]
+        elsif build.with? "scotch"
+          simple_args += ["scotch_libdir=#{Formula["scotch"].opt_lib}",
+                          "scotch_libs=-L$(scotch_libdir) -lptscotch -lptscotcherr -lscotch"]
+        end
+        if build.with? "openblas"
+          simple_args += ["blas_libdir=#{Formula["openblas"].opt_lib}",
+                          "blas_libs=-L$(blas_libdir) -lopenblas"]
+        end
+        system "make", "SHELL=/bin/bash", *simple_args
+        lib.install "libmumps_simple.dylib"
+        include.install "mumps_simple.h"
+      end
+    end
+  end
+
+  def caveats
+    s = <<-EOS.undent
+      MUMPS was built with shared libraries. If required,
+      static libraries are available in
+        #{opt_libexec}/lib
+    EOS
+    if build.without? "mpi"
+      s += <<-EOS.undent
+      You built a sequential MUMPS library.
+      Please add #{libexec}/include to the include path
+      when building software that depends on MUMPS.
+      EOS
+    end
+    s
+  end
+
+  test do
+    ENV.fortran
+    cp_r pkgshare/"examples", testpath
+    opts = ["-I#{opt_include}", "-L#{opt_lib}", "-lmumps_common", "-lpord"]
```

Comment:
should use `include` and `lib` not `opt_include` and `opt_lib` for the formula itself